File: microservices/src/main/java/by/ese/components/microservices/Registry.java
package by.ese.components.microservices;

import java.util.List;

/**
 * Microservices Registry.
 */
public interface Registry {
    String REGISTER = "/bind";
    String GET_SERVICE = "/lookup";
    String GET_SERVICES = "/list";

    RegistrationStatus register(Service service);

    Service getService(String name);

    List<Service> getServices();
}

File: microservices/src/main/java/by/ese/components/microservices/Registry.kt
package by.ese.components.microservices

/**
 * Microservices Registry.
 */
interface Registry {
    fun register(service: Service?): RegistrationStatus?
    fun getService(name: String?): Service?
    val services: List<Service?>?

    companion object {
        const val REGISTER = "/bind"
        const val GET_SERVICE = "/lookup"
        const val GET_SERVICES = "/list"
    }
}